### PR TITLE
Add video player buffering status

### PR DESCRIPTION
**Pull request by skybur**
I have added property to know whether video player is currently buffering or not. I need this to show `CircularProgressIndicator` if the video is currently buffering. I haven't tested for the IOS because I don't have the devices so if anyone can test it, please do.

### CODE REVIEW

File: packages/video_player/lib/video_player.dart [hunk 1]
```diff
@@ -230,6 +237,16 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
             buffered: values.map<DurationRange>(toDurationRange).toList(),
           );
           break;
+        case 'bufferingStart':
+          value = value.copyWith(
+            isBuffering: true,
+          );
```
Comment (mravn-google) [inline packages/video_player/lib/video_player.dart]: [nit] Here and below: we use trailing commas only when needed. And this statement easily fits on a single line:
```dart
value = value.copyWith(isBuffering: true);
```

File: packages/video_player/ios/Classes/VideoPlayerPlugin.m [hunk 2]
```diff
@@ -162,6 +172,14 @@ - (void)observeValueForKeyPath:(NSString*)path
...
       [self updatePlayingState];
     }
+  } else if (context == playbackBufferEmptyContext) {
+    if (_eventSink != nil) {
+      _eventSink(@{@"event" : @"bufferingStart"});
+    }
+  } else if (context == playbackBufferFullContext) {
+    if (_eventSink != nil) {
+      _eventSink(@{@"event" : @"bufferingEnd"});
+    }
```
Comment (mravn-google) [inline packages/video_player/ios/Classes/VideoPlayerPlugin.m]: This doesn't quite work. Running the example app on my iOS device, the circular progress indicator is activated, but then stays active throughout playback.

I guess if the video fits in the buffer, the `playbackBufferFull` property never changes. So we probably have to fire the `bufferingEnd` event also on `playbackLikelyToKeepUp`.

File: packages/video_player/lib/video_player.dart (hunk 1 above)
Comment (skybur) [reply packages/video_player/lib/video_player.dart]: Ah yeah sorry, force of habit. Will remove.

File: packages/video_player/ios/Classes/VideoPlayerPlugin.m (hunk 2 above)
Comment (skybur) [reply packages/video_player/ios/Classes/VideoPlayerPlugin.m]: Alright, gonna add that.

**Comment by mravn-google** [comment]
Published, thanks again!

**Comment by skybur** [comment]
Thanks for reviewing 😁

**Pull request merged**